ush/forecast_postdet.sh

```diff
@@ -350,27 +349,52 @@ WW3_postdet() {
   fi
 
   echo "Copying WW3 restarts for 'RUN=${RUN}' at '${restart_date}' from '${restart_dir}'"
+
   ww3_restart_file="${restart_dir}/${restart_date:0:8}.${restart_date:8:2}0000.restart.ww3"
   if [[ -s "${ww3_restart_file}" ]]; then
+    #first check if a binary restart exists, if it does set 
+    export WW3_restart_from_binary=true
     ${NCP} "${ww3_restart_file}" "${DATA}/restart.ww3" \
-      || ( echo "FATAL ERROR: Unable to copy WW3 IC, ABORT!"; exit 1 )
+          || ( echo "FATAL ERROR: Unable to copy WW3 IC, ABORT!"; exit 1 )
     first_ww3_restart_out=$(date --utc -d "${restart_date:0:8} ${restart_date:8:2} + ${restart_interval} hours" +%Y%m%d%H)
+    if [[ "${DOIAU:-NO}" == "YES" ]]; then
+      first_ww3_restart_out=$(date --utc -d "${first_ww3_restart_out:0:8} ${first_ww3_restart_out:8:2} + ${half_window} hours" +%Y%m%d%H)
+    fi 
   else
-    if [[ "${RERUN}" == "YES" ]]; then
-      # In the case of a RERUN, the WW3 restart file is required
-      echo "FATAL ERROR: WW3 restart file '${ww3_restart_file}' not found for RERUN='${RERUN}', ABORT!"
-      exit 1
-    else
-      echo "WARNING: WW3 restart file '${ww3_restart_file}' not found for warm_start='${warm_start}', will start from rest!"
-      first_ww3_restart_out=${model_start_date_current_cycle}
+    # now check to see if a NetCDF restart exists: 
+    seconds=$(to_seconds "${restart_date:8:2}0000")  # convert HHMMSS to seconds
+    ww3_restart_dest_file="ufs.cpld.ww3.r.${restart_date:0:4}-${restart_date:4:2}-${restart_date:6:2}-${seconds}.nc"
+
+    ww3_restart_file="${restart_dir}/${restart_date:0:8}.${restart_date:8:2}0000.restart.ww3.nc"
+    if [[ -s "${ww3_restart_file}" ]]; then  
+      # NetCDF restart exists: 
+      export WW3_restart_from_binary=false
+      ${NCP} "${ww3_restart_file}" "${DATA}/${ww3_restart_dest_file}" \
+             || ( echo "FATAL ERROR: Unable to copy WW3 IC, ABORT!"; exit 1 )
+      first_ww3_restart_out=$(date --utc -d "${restart_date:0:8} ${restart_date:8:2} + ${restart_interval} hours" +%Y%m%d%H)
+      if [[ "${DOIAU:-NO}" == "YES" ]]; then
+        first_ww3_restart_out=$(date --utc -d "${first_ww3_restart_out:0:8} ${first_ww3_restart_out:8:2} + ${half_window} hours" +%Y%m%d%H)
+      fi
+    else    
+      if [[ "${RERUN}" == "YES" ]]; then
+        # In the case of a RERUN, the WW3 restart file is required
+        echo "FATAL ERROR: WW3 restart file '${ww3_restart_file}' not found for RERUN='${RERUN}', ABORT!"
+        exit 1
+      else
+        # TODO: See if this will work still with new options... might not
+        echo "WARNING: WW3 restart file '${ww3_restart_file}' not found for warm_start='${warm_start}', will start from rest!"
+        first_ww3_restart_out=${model_start_date_current_cycle}
+      fi
```

Comment:
I am not sure I follow this sequence (or may be I just find it very confusing)
The first check is for a binary restart, followed by a netcdf restart.
If the netcdf restart is also not found, then a check for `RERUN` is made in the else block of the binary check.

Perhaps this can be broken down and simplified a bit.
Here is what I am proposing:
1. check for the restart file (either binary or netcdf).  the distinction is the `.nc` extension.  I assume, `.nc` is the preferred one, so check that first.
2. Regardless of `binary` or `netcdf`, the `DOIAU` block can be executed to give `first_ww3_restart_out`.
3. The `RERUN` and else block are still in flux based on the comment, so just leave them as stubs.

Reply:
checking my to-do to see if we can even start w/out a ww3 IC and cleaning up otherwise as suggested... will have updates soon.